omegaconf/_utils.py

```diff
@@ -593,6 +598,20 @@ def is_tuple_annotation(type_: Any) -> bool:
         return origin is tuple  # pragma: no cover
 
 
+def is_literal_annotation(type_: Any) -> bool:
+    origin = getattr(type_, "__origin__", None)
+    if sys.version_info >= (3, 8, 0):
+        return origin is Literal  # pragma: no cover
+    else:
+        return (
+            origin is Literal
+            or type_ is Literal
+            or (
+                "typing_extensions.Literal" in str(type_) and not isinstance(type_, str)
+            )
```

Comment:
Hmm, why is this needed? Isn't `Literal` always from `typing_extensions` in this `else` branch?

Reply:
So what's happening here is that:

python 3.8 and later has `typing.Literal` which has an `__origin__` attribute
python 3.7 `typing_extensions.Literal` has an `__origin__` attribute
python 3.6 and earlier `typing_extensions.Literal` does not have an `__origin__` attribute or any equivalent attribute

AND

```python
A = Literal[5]
A is Literal  # False
isinstance(A, Literal)  # TypeError: isinstance() arg 2 must be a type or tuple of types
issubclass(type(A), Literal)  # TypeError: issubclass() arg 2 must be a class or tuple of classes
```

so as far as I can tell, for 3.6 and earlier the only way to check for Literalness is using the string. This is gross but I don't know of a better way.

This logic is unnecessarily complicated though. How do you feel about

```python
def is_literal_annotation(type_: Any) -> bool:
    origin = getattr(type_, "__origin__", None)
    return origin is Literal or ("typing_extensions.Literal" in str(type_) and not isinstance(type_, str))
```

Reply:
There is an `is_literal_type()` function [here](https://github.com/ilevkivskyi/typing_inspect/blob/master/typing_inspect.py#L222) which also work with Python 3.6. The solution to the problem seems very stupid, you have to import a hidden class `_Literal`:

```
Python 3.6.13 |Anaconda, Inc.| (default, Jun  4 2021, 14:25:59)
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing_extensions import Literal, _Literal
>>> a = Literal[0]
>>> type(a) is Literal
False
>>> type(a) is _Literal
True
```
I have no idea why.